src/fpm/metadata/models.py

```diff
@@ -227,6 +227,23 @@ class Install(BaseModel):
     """Install library with this project"""
 
 
+class Preprocess(BaseModel):
+    """
+    .. Preprocessing settings
+
+    Preprocessing directives for the project.
+    """
+
+    macros: List[str] = []
```

Comment:
This is actually a `Union[str, List[str]]` for all elements of this class.

Reply:
so is `preprocess.macros = "HAVE_MPI"` considered legal? I thought we could only do lists